src/index.js

```diff
@@ -5,12 +5,26 @@ export default function (Alpine) {
 
       const hasDynamicTemplate = this.hasAttribute(':template')
       const hasDynamicUrl = this.hasAttribute(':url')
+      const hasDynamicStyles = this.hasAttribute(':styles')
 
-      if (hasDynamicTemplate || hasDynamicUrl) {
+      if (hasDynamicTemplate || hasDynamicUrl || hasDynamicStyles) {
         Alpine.initTree(this)
       }
 
-      const { template: componentTemplate, url: componentUrl } = this.attributes
+      const { template: componentTemplate, url: componentUrl, styles: componentStyles } = this.attributes
+
+      if (componentStyles) {
+        const styles = new CSSStyleSheet()
+        styles.insertRule(':host { display: contents; }')
+        let i = 0
+        for (const rule of [...document.styleSheets].flatMap((sheet) => [...sheet.cssRules])) {
+          if (rule instanceof CSSStyleRule && rule.selectorText === ':root') {
+            continue
+          }
+          styles.insertRule(rule.cssText, i++)
+        }
+        shadowDom.adoptedStyleSheets = [styles]
+      }
```

Comment:
This feels too much like computer speak, I think it should be more declarative. 

There must be a more modern approach that using `i++`.

Reply:
Do we even need `i++`? I just rewrote it like this and it seems to be working OK:

```js
if (componentStyles) {
  const newStyle = new CSSStyleSheet()
  const documentStyles = [...document.styleSheets].flatMap(({ cssRules }) => [...cssRules])

  for (const styleRule of documentStyles) {
    if (styleRule instanceof CSSStyleRule && styleRule.selectorText === ':root') {
      continue
    }

    newStyle.insertRule(styleRule.cssText)
  }

  shadowDom.adoptedStyleSheets = [newStyle]
}
```

Reply:
Yeah I'll run prettier on it to match the rest of the formatting 👍 

As for the index, it is now optional [`CSSStyleSheet.insertRule#parameters`](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet/insertRule#parameters) but in the past it was a mandatory param (here's the browser [compatibility list](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet/insertRule#browser_compatibility), though I don't think it'd impact a lot of people, the browsers versions are all pretty old

Reply:
If you pull down the latest it should have all the prettier updates.